### PR TITLE
SketchOperator for bloom filters

### PR DESCRIPTION
**Pull request by jiayu-google**
Add SketchOperator for bloom filters. This is for the frequency dedup… based on stratefied sketches.

### CODE REVIEW

File: src/estimators/bloom_filters.py [hunk 1]
```diff
@@ -545,3 +546,143 @@ def _denoise_one(self, sketch):
     denoised_sketch.sketch = 1 - expected_zeros / (
         1 - self._probability[1] - self._probability[0])
     return denoised_sketch
+
+
+class SketchOperator(object):
+  """Sketch operations for supporting local DP frequency dedupe."""
+
+  BAYESIAN_APPROXIMATION = "bayesian"
```
Comment (kungfucraig) [inline src/estimators/bloom_filters.py:554]: Per offline discussion. Move sketch operators to "bloom_filter_sketch_operators.py" and create a base class SketchOperator with two derived class, one each for bayesian and expectation.

File: src/estimators/bloom_filters.py [hunk 2]
```diff
@@ -545,3 +546,143 @@ def _denoise_one(self, sketch):
...
+    """Construct an AnyDistributionBloomFilter sketch operator.
+
+    Args:
+      estimation_method: distribution of bit probabilities in ADBF. That is,
+        method to specify in FirstMomentEstimator.
+      approximation_method: method used in approximating the intersection and
+        difference operations between two AnyDistributionBloomFilters.
+      threshold: a threshold to avoid numerical errors.
+    """
+    self.estimation_method = estimation_method
```
[the remark this answers is not in the document]
Comment (jiayu-google) [reply src/estimators/bloom_filters.py]: Updated. Please review!

File: src/estimators/bloom_filters.py (hunk 1 above)
Comment (jiayu-google) [reply src/estimators/bloom_filters.py]: Updated. Please review!

**Review by AaronRanAn — APPROVED** [review]
LGTM!

**Review by shengma-1 — APPROVED** [review]

**Review by kungfucraig — APPROVED** [review]

**Review by huangxichen1 — APPROVED** [review]

**Pull request merged**